boot/bootstate20.go

```diff
@@ -233,6 +233,123 @@ func (bks *extractedRunKernelImageBootloaderKernelState) setNextKernel(sn snap.P
 	return nil
 }
 
+// envRefExtractedKernelBootloaderKernelState implements bootloaderKernelState20 for
+// bootloaders that only support using bootenv and i.e. don't support
```

Comment:
there is a termology confusion here, bootenv here is really bootloader env. boot env is any boot state including symlinks